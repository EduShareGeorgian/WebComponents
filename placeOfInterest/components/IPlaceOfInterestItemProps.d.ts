export interface IPlaceOfInterestItemProps {
    name: string
    type: string
    detailsLink?: URL
    mapLink?: URL
    locationId: string
    hoursDescription: string
    hasCamera: boolean
    onDetailsLinkSelected?: (url: URL) => void;
    onCameraLinkSelected?: (url: URL) => void;
    /** Callback for when an items Map link is selected/clicked */
    onMapLinkSelected?: (url: URL) => void;
}