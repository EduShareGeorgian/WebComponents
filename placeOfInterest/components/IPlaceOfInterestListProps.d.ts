export interface IPlaceOfInterestListProps {
    items: any[];
    onDetailsLinkSelected?: (url: URL) => void;
    onCameraLinkSelected?: (url: URL) => void;
    /** Callback for when an items Map link is selected/clicked */
    onMapLinkSelected?: (url: URL) => void;
}