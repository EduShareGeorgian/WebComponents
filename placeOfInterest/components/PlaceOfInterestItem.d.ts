/// <reference types="react" />
import * as React from 'react';
import { IPlaceOfInterestItemProps } from './IPlaceOfInterestItemProps';
export default class PlaceOfInterestItem extends React.Component<IPlaceOfInterestItemProps, any> {
    constructor(props: IPlaceOfInterestItemProps);
    protected _onMapLinkClick(event: React.MouseEvent<HTMLElement>): void;
    protected _onDetailsLinkClick(event: React.MouseEvent<HTMLElement>): void;
    render(): JSX.Element;
}
export { PlaceOfInterestItem, IPlaceOfInterestItemProps };
