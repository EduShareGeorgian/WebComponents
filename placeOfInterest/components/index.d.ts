import {IPlaceOfInterestItemProps} from "./IPlaceOfInterestItemProps"
import {IPlaceOfInterestListProps} from "./IPlaceOfInterestListProps"
import PlaceOfInterestItem from "./PlaceOfInterestItem.d"
import PlaceOfInterestList from "./PlaceOfInterestList.d"

export { PlaceOfInterestItem, PlaceOfInterestList, IPlaceOfInterestListProps, IPlaceOfInterestItemProps }
