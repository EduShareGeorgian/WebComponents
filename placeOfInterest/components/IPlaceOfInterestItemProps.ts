export interface IPlaceOfInterestItemProps {
    name: string,
    placeLink?: URL,
    mapLink?: URL,
    locationId: string,
    hoursDescription: string,
    hasCamera: boolean,
    launchMap?: Function,
    launchPlaceDetails?: Function,
    launchCameraView?: Function
}