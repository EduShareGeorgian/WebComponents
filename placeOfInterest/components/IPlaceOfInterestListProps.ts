export interface IPlaceOfInterestListProps {
    items: any[];
    onDetailsLinkSelected?: (item?: any, index?: number, ev?: React.MouseEvent<HTMLElement>) => void;
    onCameraLinkSelected?: (item?: any, index?: number, ev?: React.MouseEvent<HTMLElement>) => void;
    /** Callback for when an items Map link is selected/clicked */
    onMapLinkSelected?: (item?: any, index?: number, ev?: React.MouseEvent<HTMLElement>) => void;
}