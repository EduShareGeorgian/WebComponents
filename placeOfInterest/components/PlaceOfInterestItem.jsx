import React from 'react';


class PlaceOfInterestItem extends React.Component {

  render() {
    return (
        <li className="placeOfInterestItem ms-Grid-row">
          <div className="ms-Grid-col ms-u-sm3"><a href={this.props.placeLink}>{this.props.name}</a></div>
          <a target="_blank" href={this.props.mapLink} className="mapLink" onClick={this.props.launchMap.bind(this)}>
            {this.props.hasCamera &&
              <div class="foodServicesHoursRestaurant ms-Grid-col ms-u-sm1">
                <i class="ms-Icon ms-Icon--Camera" aria-hidden="true"></i>
              </div>
            }
            <div className="ms-Grid-col ms-u-sm1" aria-hidden="true">&nbsp;</div>
            <div className="ms-Grid-col ms-u-sm1">
              <i className="ms-Icon ms-Icon--POI" aria-hidden="true"></i>
            </div>
            <div className="ms-Grid-col ms-u-sm2">{this.props.locationId}</div>
          </a>
          <div className="ms-Grid-col ms-u-sm5">{this.props.hoursDescription}</div>
        </li>)
  }

}

export {PlaceOfInterestItem}