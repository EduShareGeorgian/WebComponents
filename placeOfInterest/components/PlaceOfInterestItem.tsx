import * as React from 'react';
import {IPlaceOfInterestItemProps} from './IPlaceOfInterestItemProps'

export default class PlaceOfInterestItem extends React.Component<IPlaceOfInterestItemProps, any> {
    constructor(props: IPlaceOfInterestItemProps) {
        super(props);
        this._onMapLinkClick = this._onMapLinkClick.bind(this);
        this._onDetailsLinkClick = this._onDetailsLinkClick.bind(this);
    }

    protected _onMapLinkClick(event: React.MouseEvent<HTMLElement>) {
        event.preventDefault();
        event.stopPropagation();
        let {onMapLinkSelected} = this.props;
        if (onMapLinkSelected && this.props.mapLink) {
            onMapLinkSelected(this.props.mapLink);
        }
    }

    protected _onDetailsLinkClick(event: React.MouseEvent<HTMLElement>) {
        event.preventDefault();
        event.stopPropagation();
        let {onDetailsLinkSelected} = this.props;
        if (onDetailsLinkSelected && this.props.detailsLink) {
            onDetailsLinkSelected(this.props.detailsLink);
        }
    }

    public render() {
        return (
            <li className="placeOfInterestItem ms-Grid-row">
                <div className="ms-Grid-col ms-u-sm3"><a href={this.props.detailsLink && this.props.detailsLink.href} className="detailsLink"
                                                         onClick={this._onDetailsLinkClick}>{this.props.name}</a></div>
                <a target="_blank" href={this.props.mapLink && this.props.mapLink.href} className="mapLink" onClick={this._onMapLinkClick}>
                    {this.props.hasCamera &&
                    <div className="ms-Grid-col ms-u-sm1">
                        <i className="ms-Icon ms-Icon--Camera" aria-hidden="true"></i>
                    </div>
                    }
                    <div className="ms-Grid-col ms-u-sm1" aria-hidden="true">&nbsp;</div>
                    <div className="ms-Grid-col ms-u-sm1">
                        <i className="ms-Icon ms-Icon--POI" aria-hidden="true"></i>
                    </div>
                    <div className="ms-Grid-col ms-u-sm2">{this.props.locationId}</div>
                </a>
                <div className="ms-Grid-col ms-u-sm5">{this.props.hoursDescription}</div>
            </li>)
    }

}

export {PlaceOfInterestItem, IPlaceOfInterestItemProps}