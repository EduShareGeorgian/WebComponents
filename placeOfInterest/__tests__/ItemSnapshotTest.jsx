import {PlaceOfInterestItem} from "../components/PlaceOfInterestItem";
import React from "react"
import renderer from "react-test-renderer"


describe('components', () => {
  describe('PlaceOfInterestItem', () => {
    const defaultProps = () => {
      return {
        name: "Chartwells",
        placeLink: "https://subway.com/home",
        mapLink: "https://maps.google.ca/?q=Chartwells",
        locationId: "E109",
        hoursDescription: "7:30am - 2:00pm",
        hasCamera: false,
        launchMap: jest.fn(),
        launchPlaceDetails: jest.fn(),
        launchCameraView: jest.fn()
      }
    };
    it('renders correctly without Camera Icon', () => {
      const tree = renderer.create(
          <PlaceOfInterestItem {...defaultProps()}></PlaceOfInterestItem>
      ).toJSON();
      expect(tree).toMatchSnapshot("WithoutCamera");
    })
    it('renders correctly with Camera Icon', () => {
      const props = defaultProps();
      props.hasCamera = true;
      const tree = renderer.create(
          <PlaceOfInterestItem {...props}></PlaceOfInterestItem>
      ).toJSON();
      expect(tree).toMatchSnapshot("WithCamera");
    });
  });
});
