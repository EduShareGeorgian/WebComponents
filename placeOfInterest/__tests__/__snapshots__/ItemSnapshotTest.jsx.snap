// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithCamera 1`] = `
<li
  className="placeOfInterestItem ms-Grid-row"
>
  <div
    className="ms-Grid-col ms-u-sm3"
  >
    <a
      className="detailsLink"
      href="https://subway.com/home"
      onClick={[Function]}
    >
      Chartwells
    </a>
  </div>
  <a
    className="mapLink"
    href="https://maps.google.ca/?q=Chartwells"
    onClick={[Function]}
    target="_blank"
  >
    <div
      className="ms-Grid-col ms-u-sm1"
    >
      <i
        aria-hidden="true"
        className="ms-Icon ms-Icon--Camera"
      />
    </div>
    <div
      aria-hidden="true"
      className="ms-Grid-col ms-u-sm1"
    >
       
    </div>
    <div
      className="ms-Grid-col ms-u-sm1"
    >
      <i
        aria-hidden="true"
        className="ms-Icon ms-Icon--POI"
      />
    </div>
    <div
      className="ms-Grid-col ms-u-sm2"
    >
      E109
    </div>
  </a>
  <div
    className="ms-Grid-col ms-u-sm5"
  >
    7:30am - 2:00pm
  </div>
</li>
`;

exports[`WithoutCamera 1`] = `
<li
  className="placeOfInterestItem ms-Grid-row"
>
  <div
    className="ms-Grid-col ms-u-sm3"
  >
    <a
      className="detailsLink"
      href="https://subway.com/home"
      onClick={[Function]}
    >
      Chartwells
    </a>
  </div>
  <a
    className="mapLink"
    href="https://maps.google.ca/?q=Chartwells"
    onClick={[Function]}
    target="_blank"
  >
    <div
      aria-hidden="true"
      className="ms-Grid-col ms-u-sm1"
    >
       
    </div>
    <div
      className="ms-Grid-col ms-u-sm1"
    >
      <i
        aria-hidden="true"
        className="ms-Icon ms-Icon--POI"
      />
    </div>
    <div
      className="ms-Grid-col ms-u-sm2"
    >
      E109
    </div>
  </a>
  <div
    className="ms-Grid-col ms-u-sm5"
  >
    7:30am - 2:00pm
  </div>
</li>
`;
